front/src/modules/pathfinding/utils.ts

```diff
@@ -177,3 +177,29 @@ export const isStation = (chCode: string): boolean =>
   chCode === 'BV' || chCode === '00' || chCode === '';
 
 export const isPathStepInvalid = (step: PathStep | null): boolean => step?.isInvalid || false;
+
+/**
+ * Ensure that the array doesn't have any duplicated operational points by merging the duplicates.
+ * They usually can be differentiated by the track name.
+ * @param operationalPoints the list of operational points
+ * @param getDuplicatedOP a function that returns a key base on some op properties
+ * @param mergeOP a function checking that previous and next op differ by one property and merge this property into the previous op
+ * @returns the updated list with unique operational points
+ */
+export function getUniqueOperationalPoints<T>(
+  operationalPoints: T[],
+  extractKeyFromOp: (operationalPoint: T) => string,
```

Comment:
```suggestion
  getDuplicatedOP: (operationalPoint: T) => string,
```
To comply with function documentation

Reply:
Actually I asked @Akctarus to change this param name but forgot to change it in the documentation proposal :)

Reply:
fixed